synapse/tools/migrate_010.py

```diff
@@ -0,0 +1,715 @@
+
+        if formname in self.form_renames:
+            formname = self.form_renames[formname]
+
+        assert '' not in props
+
+        retn = ((formname, pk), {'props': props})
+        if tags:
+            retn[1]['tags'] = tags
+
+        return retn
+
+    def convert_ps_image(self, formname, props):
+        '''
+        Transform ps:image into a file:ref
+        '''
+        person, file = self.convert_sepr(formname, formname, props[formname], props)
+        return (file, ('ps:person', person))
+
+    primary_prop_special = {
+        'inet:web:post': just_guid,
+        'it:dev:regval': just_guid,
+        'file:bytes': convert_file_bytes,
+        'inet:udp4': convert_inet_xxp_primary,
+        'inet:udp6': convert_inet_xxp_primary,
+        'inet:tcp4': convert_inet_xxp_primary,
+        'inet:tcp6': convert_inet_xxp_primary,
+        'ps:image': convert_ps_image,
+        'file:base': check_file_base_primary,
+    }
+
+    def handle_seen(self, propname, propval, newprops):
+        '''
+        :seen:min, :seen:max -> .seen = (min, max)
+        '''
+        seenmin, seenmax = newprops.get('.seen', (None, None))
+        if propname.endswith(':min'):
+            seenmin = propval
+        if propname.endswith(':max'):
+            seenmax = propval
+        if seenmax is None:
+            seenmax = seenmin + 1
+        if seenmin is None:
+            seenmin = seenmax - 1
+        return '.seen', (seenmin, seenmax)
+
+    def ip_with_port_to_server(self, formname, propname, typename, val, props):
+        port = props.get(formname + ':port')
+        if propname[-1] == '4':
+            if port is not None:
+                val = (val << 16) + port
+            addrport = self.core.getTypeRepr('inet:srv4', val)
+        else:
+            if port is None:
+                addrport = '[' + val + ']'
+            else:
+                addrport = '[%s]:%s' % (val, port)
+        return propname, '%s://%s' % (formname[-3:], addrport)
+
+    subprop_special = {
+        'inet:exec:url:ipv4': ipv4_to_client,
+        'inet:exec:url:ipv6': ipv6_to_client,
+        'inet:web:logon:ipv4': ipv4_to_client,
+        'inet:web:logon:ipv6': ipv6_to_client,
+        'inet:web:action:ipv4': ipv4_to_client,
+        'inet:web:action:ipv6': ipv6_to_client,
+        'inet:web:acct:signup:ipv4': ipv4_to_client,
+        'inet:dns:look:ipv4': ipv4_to_client,
+        'inet:dns:look:ipv6': ipv6_to_client,
+        'it:exec:bind:tcp:ipv4': ip_with_port_to_server,
+        'it:exec:bind:tcp:ipv6': ip_with_port_to_server,
+        'it:exec:bind:udp:ipv4': ip_with_port_to_server,
+        'it:exec:bind:udp:ipv6': ip_with_port_to_server,
+    }
+
+def main(argv, outp=None):  # pragma: no cover
+    p = argparse.ArgumentParser(description='''Command line tool to export a Synapse Cortex v. 0.0.5 to a mpk file
+ for importation into a 0.1.0 cortex''')
+    p.add_argument('cortex', help='telepath URL for a cortex to be dumped')
```

Comment:
This cannot be a true telepath url since a remote cortex cannot be migrated since we touch the store object.